### PR TITLE
Feat: Add support for viewmatcher

### PR DESCRIPTION
**Pull request by ccharnkij**
## Change list

Per [appium/appium#13747](https://github.com/appium/appium/issues/13747), add support for viewmatcher along with test for datamatcher 
 
## Types of changes

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Similar to dataMatcher, provides a functionality for viewMatcher. Also I noticed that dataMatcher had no test, so I added one for dataMatcher as well.

### CODE REVIEW

File: src/main/java/io/appium/java_client/MobileBy.java [hunk 1]
```diff
@@ -106,6 +106,17 @@ public static By androidDataMatcher(final String dataMatcherString) {
         return new ByAndroidDataMatcher(dataMatcherString);
     }
 
+    /**
+     * This locator strategy is only available in Espresso Driver mode.
+     * @param viewMatcherString is a valid class chain locator string.
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/MobileBy.java]: class chain?

**Comment by mykola-mokhnach** [comment]
Please sign the CLA, so we could merge the PR. Thanks

File: src/main/java/io/appium/java_client/MobileBy.java (hunk 1 above)
Comment (ccharnkij) [reply src/main/java/io/appium/java_client/MobileBy.java]: The sin of copy & paster...I got it from the dataMatcher. I'll add a new commit in real quick, for the dataMatcher too. My apologies...

**Comment by ccharnkij** [comment]
I believe the cla check is good.

**Comment by mykola-mokhnach** [comment]
> I believe the cla check is good.

the bot verifies the email address set to your local git config and compares it with the one used for CLA signing. Just make sure you created the PR using the correct local email

**Comment by mykola-mokhnach** [comment]
The PR is fine, but I unfortunately cannot merge it unless the bot confirms the CLA is signed. Make sure the email address in your local github config is correct and recreate the PR

**Comment by ccharnkij** [comment]
@mykola-mokhnach Doesn't this mean that the bot successfully confirms CLA is signed?

Screen Shot 2020-01-16 at 12 35 48 PM

**Comment by mykola-mokhnach** [comment]
yes, this looks good. Although it is still weird why the bot didn't change the first comment. @jlipps do you know what the expected behaviour is?

**Comment by jlipps** [comment]
Yeah expected behavior is only to approve the fixed commit. As long as the PR is green we should be good to go

**Comment by mykola-mokhnach** [comment]
ok, thanks for checking this @jlipps

**Pull request merged**